src/vfd/sriov.c

```diff
@@ -1215,6 +1212,14 @@ bnxt_vf_msb_event_callback(uint8_t port_id, enum rte_eth_event_type type, void *
 
 		/* Verify */
 	}
+	if (mbox_type == HWRM_CFA_L2_FILTER_ALLOC)
+		vf_link_status[port_id][vf] = 1;
+	/* Most of the FW commands during cleanup are ALLOWED in bnxt_ethdev.c
+	 * So we may not see them here at all. So to mark the link down
+	 * first do an ifdown. Otherwise it will be tricky.
+	 */
+	if (mbox_type == HWRM_TUNNEL_DST_PORT_FREE)
```

Comment:
And this one is absolutely the wrong one.  I think we need some input from arch on this issue.